src/app/components/site/Homepage.tsx

```diff
@@ -108,20 +115,32 @@ export const Homepage = () => {
                 <hr/>
             </Container>}
 
-            {!user?.loggedIn && <section id="why-sign-up" className="row sign-up-tabs">
+            {!user?.loggedIn && <><section id="why-sign-up" className="row sign-up-tabs">
                 <Container>
                     <Col className="pb-5 pt-4 pattern-04">
                         <h2 className="text-center mb-5">Why sign up?</h2>
                         <WhySignUpTabs/>
                     </Col>
                 </Container>
-            </section>}
+            </section>
+
+            <section data-testid="promo-content" id="promo-content" className="row bg-primary pattern-05">
+                <ShowLoading
```

Comment:
I may not be understanding this correctly, but how come a `ShowLoading` component is needed here? It does not look like `promoItem` ever changes, so I would have thought that you don't need a loading helper.

Reply:
Maybe it's just an alternative way to say: if there is no promo item, just display the loading component. Otherwise, display PromoContent.

Reply:
This was a funny one, I used the existing ShowLoading component that's used elsewhere (such as FeaturedNewsItem), to let it show Loading until the API request is complete. But it still needed the conditional on line 132 otherwise typescript was unhappy.  

Do you mean it would be better to ignore the ShowLoading component and just do 

`{promoItem ? <PromoContent item={promoItem} /> : <defaultPlaceholder />}`

defaultPlaceholder is the Loading spinner in /handlers/ShowLoading.tsx

I guess though that would eternally show Loading if the API request failed to fetch the content for any reason.

Reply:
Yes, that's exactly what I meant, but I must be misunderstanding something. From a first glance, it looked like this component has no state or side effects (apart from the document's title). That's why I found the use of `ShowLoading` weird: if there is no state or `useEffect` or props, then once the component is loaded, nothing will change, so why do we need this `ShowLoading` component? But I just realised that this file imports 3 things from `state`, so maybe this is using Redux in a way that I'm not familiar with.

TL;DR: ignore me 😂